test/unit/select_plan_test.lua

```diff
@@ -339,3 +339,203 @@ g.test_table_count = function()
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 1 }} },
+        conditions = {{ '<', 'id', 2 }},
+        -- Expected values.
+        sharding_key = nil,
+    },
+    extract_field_sharding_key_from_its_non_unique_single_field_secondary_index_condition = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 5 }} },
+        conditions = {{ '==', 'age', 42 }},
+        -- Expected values.
+        sharding_key = { 42 },
+    },
+    extract_field_sharding_key_from_its_multiple_fields_secondary_index_condition = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }} },
+        conditions = {{ '==', 'full_name', { 'John',  'Doe' } }},
+        -- Expected values.
+        sharding_key = { 'John' },
+    },
+    extract_table_sharding_key_from_two_conditions = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }, { fieldno = 5 }} },
+        conditions = {{ '==', 'name', 'John' }, { '==', 'age', 42 }},
+        -- Expected values.
+        sharding_key = { 'John', 42 },
+    },
+    extract_table_sharding_key_from_two_index_conditions = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }, { fieldno = 5 }} },
+        conditions = {{ '==', 'full_name', { 'John',  'Doe' } }, { '==', 'age', 42 }},
+        -- Expected values.
+        sharding_key = { 'John', 42 },
+    },
+    extract_table_sharding_key_from_eq_an_ge_conditions_returns_nil = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }, { fieldno = 5 }} },
+        conditions = {{ '==', 'name', 'John' }, { '>=', 'age', 42 }},
+        -- Expected values.
+        sharding_key = nil,
+    },
+    extract_table_sharding_key_from_partial_conditions = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 1 }, { fieldno = 4 }} },
+        conditions = {{ '==', 'full_name', { nil,  'Doe' } }, { '==', 'id', 1 }},
+        -- Expected values.
+        sharding_key = { 1, 'Doe' },
+    },
+    extract_table_sharding_key_from_partial_conditions_with_nil_and_non_nil_for_same_value_in_different_conditions = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }, { fieldno = 5 }} },
+        conditions = {{ '==', 'full_name', { 'John',  'Doe' } }, { '==', 'name_id', { nil, 1 } }},
+        -- Expected values.
+        sharding_key = nil,
+    },
+}
+
+for name, case in pairs(extract_sharding_key_from_conditions_cases) do
+    g[('test_%s'):format(name)] = function()
+        local space_indexes = box.space.customers.index
+        local space_format = box.space.customers:format()
+        local fieldno_map = utils.get_format_fieldno_map(space_format)
+        local conditions = compare_conditions.parse(case.conditions)
+
+        local sharding_key = select_plan.internal.extract_sharding_key_from_conditions(conditions, case.sharding_index,
```

Comment:
May be to create a shortcut for a function?

Reply:
Used a shortcut

Reply:
Thanks!